plugins/modules/nsupdate.py

```diff
@@ -467,10 +467,8 @@ def ttl_changed(self):
         if lookup.rcode() != dns.rcode.NOERROR:
             self.module.fail_json(msg='Failed to lookup TTL of existing matching record.')
 
-        if self.module.params['type'] == 'NS':
-            current_ttl = lookup.answer[0].ttl if lookup.answer else lookup.authority[0].ttl
-        else:
-            current_ttl = lookup.answer[0].ttl
+        current_ttl = lookup.answer[0].ttl if lookup.answer else lookup.authority[0].ttl
```

Comment:
I don't know whether using the SOA TTL as a backup is a correct, but this should definitely help against the reported problems. The worst that can happen is that the module will always report changed instead of failing, which is already a big improvement.